ansible/roles/test/tasks/copp.yml

```diff
@@ -28,6 +25,36 @@
       copy: src=roles/test/files/ptftests dest=/root
       delegate_to: "{{ ptf_host }}"
 
+# Add case for Force10-S6100 since this platform has 64 ports
+# TODO: Generalize the code so that it fits for any topologies
+    - set_fact:
+        test_port_index: "3"
+        test_port_src_ip: "10.0.0.7"
+        test_port_peer_ip: "10.0.0.6"
+        test_port_alias: "Ethernet12"
+      when: minigraph_hwsku != 'Force10-S6100'
+
+    - set_fact:
+        test_port_index: "1"
+        test_port_src_ip: "10.0.0.1"
+        test_port_peer_ip: "10.0.0.0"
+        test_port_alias: "Ethernet1"
+      when: minigraph_hwsku == 'Force10-S6100'
+
+    - block:
+      - name: Stop the original ptf_nn_agent thread started by supervisord
+        shell: supervisorctl stop ptf_nn_agent
+        vars:
+          ansible_shell_type: docker
+          ansible_python_interpreter: docker exec -i syncd python
+
+      - name: Start ptf_nn_agent on DUT
+        shell: /usr/bin/python /opt/ptf_nn_agent.py --device-socket 1@tcp://0.0.0.0:10900 -i 1-{{test_port_index}}@{{test_port_alias}} --set-iface-rcv-buffer=109430400 &
+        vars:
+          ansible_shell_type: docker
+          ansible_python_interpreter: docker exec -i syncd python
+      when: minigraph_hwsku == 'Force10-S6100'
+
```

Comment:
Can you please overwrite supervisor configuration and start it using supervisor?

Reply:
Because we have a container with working configuration which is expected from other test cases. You stop the previous configuration, run your ad-hoc configuration and keep it.
The other test could crash because of it.
So I think it's better to have a list of supervisor configuration templates for tests. If you need your own configuration, you can generate it, put into /etc/supervisor/conf.d and reload supervisor.
Later you can regenerate common configuration and reload supervisor again.

Reply:
Because we have a container with working configuration which is expected from other test cases. You stop the previous configuration, run your ad-hoc configuration and keep it.
The other test could crash because of it.
So I think it's better to have a list of supervisor configuration templates for tests. If you need your own configuration, you can generate it, put into /etc/supervisor/conf.d and reload supervisor.
Later you can regenerate common configuration and reload supervisor again.